### ISSUE TITLE
Date AddYears - Condition Operator ?

### ISSUE THREAD

**Issue by vladsch** [bridgedotnet/Bridge#2963]
### Steps To Reproduce

https://deck.net/7340f7e0a8df1a1a6540bc317bd2adea

```c#
public class Program
{
    public static void Main()
    {
        var d = new DateTime(2017, 1, 2);

        bool b = false;
        var d1 = d.AddYears(b ? -1 : 1);
        // Should be new DateTime(2018, 1, 2)
        Console.WriteLine(d1);


        b = true;
        var d2 = d.AddYears(b ? -1 : 1);
        // Should be new DateTime(2016, 1, 2)
        Console.WriteLine(d2);        
    }
}
```

### Expected Result
```
1/2/2018 00:00:00
1/2/2016 00:00:00
```

### Actual Result
```
01/02/-1 00:00:00
01/02/-1 00:00:00
```

## See Also

* https://forums.bridge.net/forum/bridge-net-pro/bugs/4574-date-addyears-condition-operator
* DateTime.MinValue is not UTC #2952
* DateTime.ToUniversalTime does not correctly set timezone #2929
* Unexpected results of ToLocalTime() / ToUniversalTime() #2524

**Comment by geoffreymcgill** [comment]
This defect has been fixed in #2945 and the fix will be included in the next public release.

**Comment by michaelcheers** [comment]
I think more importantly this is a problem in templates:
https://deck.net/10d69a156c35e2d9171e83c04752838c/third
